### PR TITLE
Packaging fixes

### PR DESCRIPTION
**Pull request by clebergnu**
This PR replaces the custom usage of "data_files" parameter to setuptools, containing locations that start with `/usr/share/`, for Python standard way of packaging data files.

This means there are a lot of files moved around in this PR, but untouched with regards to their contents.

It fixes a long standing bug that was not being covered by our deployment checks on Cirrus CI, and that we've finally found out that would only affect installations via wheels (other pip and setup.py installations worked fine).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -22,55 +22,6 @@
...
-    for data_file_dir in data_files_dirs:
-        for root, dirs, files in os.walk(data_file_dir):
-            for subdir in dirs:
-                rt = root.split('/')
-                rt.append(subdir)
-                data_files += add_files(rt)
-
-    return data_files
-
-
```
Comment (beraldoleal) [inline setup.py:73]: Nice to see this being removed from setup.py and added to Manifest. +1

File: virttest/data_dir.py [hunk 2]
```diff
@@ -18,28 +20,14 @@
...
-else:
-    _ROOT_PATH = os.path.realpath(os.path.dirname(os.path.dirname(__file__)))
-
-ROOT_DIR = os.path.abspath(_ROOT_PATH)
-BASE_BACKEND_DIR = os.path.join(ROOT_DIR, 'backends')
-DATA_DIR = os.path.join(data_dir.get_data_dir(), 'avocado-vt')
-SHARED_DIR = os.path.join(ROOT_DIR, 'shared')
-DEPS_DIR = os.path.join(ROOT_DIR, 'shared', 'deps')
+BASE_BACKEND_DIR = pkg_resources.resource_filename('virttest', 'backends')
+TEST_PROVIDERS_DIR = pkg_resources.resource_filename('virttest', 'test-providers.d')
```
Comment (beraldoleal) [inline virttest/data_dir.py:24]: Makes sense! +1

**Review by beraldoleal — APPROVED** [review]
Besides the LGTM report, this looks fine.

File: setup.py [hunk 3]
```diff
@@ -22,55 +22,6 @@
 VERSION = open('VERSION', 'r').read().strip()
 
 
-def __is_virtual_env():
-    return (hasattr(sys, 'real_prefix') or (hasattr(sys, 'base_prefix') and
```
Comment (beraldoleal) [inline setup.py:26]: Looks like sys is not being used anymore, so we can remove the import.

File: setup.py [hunk 4]
```diff
@@ -22,55 +22,6 @@
...
-
-def get_data_files():
-    def add_files(level=[]):
-        installed_location = ['usr', 'share', 'avocado-plugins-vt']
-        installed_location += level
-        level_str = '/'.join(level)
-        if level_str:
-            level_str += '/'
-        file_glob = '%s*' % level_str
-        files_found = [path for path in glob.glob(file_glob) if
```
Comment (beraldoleal) [inline setup.py:57]: Same for glob.

**Review by beraldoleal — CHANGES_REQUESTED** [review]
There are some unused imports.

File: setup.py [hunk 5]
```diff
@@ -22,55 +22,6 @@
...
-    if virtual_path is None:
-        virtual_path = system_path
-    if __is_virtual_env():
-        if virtual_path is None:
-            virtual_path = []
-        return os.path.join(*virtual_path)
-    else:
-        if system_path is None:
-            system_path = []
-        return os.path.join(*(['/'] + system_path))
```
Comment (willianrampazzo) [inline setup.py:46]: os import is also not used anymore.

File: setup.py (hunk 3 above)
Comment (clebergnu) [reply setup.py:26]: `sys` is used on `if sys.version[:2]`...

**Comment by clebergnu** [comment]
Closing in favor of #2808

**Pull request closed without merging**